webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java

```diff
@@ -4256,5 +4257,127 @@ private void setToolProperties(ToolConfiguration tool, String propList) {
             }
         }
     }
+    /**
+     * Copy the lessons pages from a site to another one. Optionally, the existing lessons pages can be deleted in destination.
+     * Links to resources are preserved and updated to the new site, but their copy is not included. copySiteContent can be used for that.
+     * Lessons pages copy does not work in copySiteContent method, neither in copySiteContentForTool.
+     * This logic could be added to copySiteContent to cover all cases, but unfortunately we were in a hurry.
+     *
+     * @param 	sessionid 			the id of a valid session
+     * @param 	sourcesiteid		the id of the source site
+     * @param 	destinationsiteid 	the id of the destiny site
+     * @param 	cleanUp				delete the existing Lessons pages (true) or preserve them (false)
+     * @return						success or exception message
+     */
+    public String copySiteLessonsPages(
+    		@WebParam(name = "sessionid", partName = "sessionid") @QueryParam("sessionid") String sessionid,
+    		@WebParam(name = "sourcesiteid", partName = "sourcesiteid") @QueryParam("sourcesiteid") String sourcesiteid,
+    		@WebParam(name = "destinationsiteid", partName = "destinationsiteid") @QueryParam("destinationsiteid") String destinationsiteid,
+    		@WebParam(name = "cleanUp", partName = "cleanUp") @QueryParam("cleanUp") boolean cleanUp)
+    {
+
+    	String toolid="sakai.lessonbuildertool";
+    	Session session = establishSession(sessionid);
+    	Set<String> toolsCopied = new HashSet<String>();
+    	Map transversalMap = new HashMap();
+
+    	try
+    	{
+    		//check if both sites exist
+    		Site site = siteService.getSite(sourcesiteid);
+    		site = siteService.getSite(destinationsiteid);
+
+    		// If not admin, check maintainer membership in the source site
+    		if (!securityService.isSuperUser(session.getUserId()) && !securityService.unlock(SiteService.SECURE_UPDATE_SITE, site.getReference()))
+    		{
+    			LOG.warn("WS copyResources(): Permission denied. Must be super user to copy a site in which you are not a maintainer.");
+    			throw new RuntimeException("WS copyResources(): Permission denied. Must be super user to copy a site in which you are not a maintainer.");
+    		}
+
+    		Map<String,String> entityMap = transferCopyEntitiesForLessonsPages(toolid, sourcesiteid, destinationsiteid, cleanUp);
+    		if(entityMap != null)
+    		{
+    			transversalMap.putAll(entityMap);
+    		}
+
+    		updateEntityReferences(toolid, sourcesiteid, transversalMap, site);
+    	}
+    	catch (Exception e)
+    	{
+    		LOG.error("WS copySiteContent(): " + e.getClass().getName() + " : " + e.getMessage(), e);
+    		return e.getClass().getName() + " : " + e.getMessage();
+    	}
+    	return "success";
+    }
 
+    protected Map transferCopyEntitiesForLessonsPages(String toolId, String fromContext, String toContext, boolean cleanup)
```

Comment:
Is this method really needed this looks like its been copied can you just use the original. Is it ideal to create a separate transferCopyEntitiesFor<insert tool name here>?